src/s1reader/s1_annotation.py

```diff
@@ -0,0 +1,343 @@
+                str_elem=elem.find(name_vector).text
+                list_out[i]=int(str_elem)
+
+        elif str_type=='scalar_float':
+            for i,elem in enumerate(element_to_parse):
+                str_elem=elem.find(name_vector).text
+                list_out[i]=float(str_elem)
+
+        elif str_type=='vector_int':
+            for i,elem in enumerate(element_to_parse):
+                str_elem=elem.find(name_vector).text
+                list_out[i]=np.array([int(strin) for strin in str_elem.split()])
+
+        elif str_type=='vector_float':
+            for i,elem in enumerate(element_to_parse):
+                str_elem=elem.find(name_vector).text
+                list_out[i]=np.array([float(strin) for strin in str_elem.split()])
+
+        elif str_type=='str':
+            list_out=element_to_parse[0].find(name_vector).text
+
+        else:
+            raise ValueError(f'Cannot recognize the type of the element: {str_type}')
+
+        return list_out
+
+
+    @property
+    def ipf_version(self):
+        '''return the string for IPF version.'''
+
+        if self._ipf_version is None:
+            if os.path.isfile(self.path_annotation):
+                str_safe=os.path.basename(self.path_annotation).replace('.zip','')
+                if self.path_annotation.endswith('.zip'):
+                    zip_in=zipfile.ZipFile(self.path_annotation,'r')
+                    str_manifest=zip_in.read(f'{str_safe}.SAFE/manifest.safe').decode()
+            elif os.path.isdir(self.path_annotation):
+                with open(f'{self.path_annotation}/manifest.safe','r',encoding='utf-8') as manifest_in:
+                    str_manifest=manifest_in.read()
+            else:
+                raise ValueError(f'Cannot find the manifest.safe file using path_annotation: {self.path_annotation}')
+
+            lines_manifest=str_manifest.split('\n')
+            for line_manifest in lines_manifest:
+                if '<safe:software ' in line_manifest:
+                    self._ipf_version=line_manifest.split('version=')[1].replace('"','').replace('/>','')
+                    break
+
+        return self._ipf_version
+
+
+
+class Calibration(Annotation):
+    '''Reader for Calibration Annotation Data Set (CADS)
+    '''
+    def __init__(self,path_annotation,polarization='VH',subswath=1):
+        super().__init__(path_annotation,polarization,subswath)
+        self.kind='calibration'
+
+        #on-the-fly attributes
+        self._list_azimuth_time=None
+        self._list_line=None
+        self._list_pixel=None
+        self._list_sigma_nought=None
+        self._list_beta_nought=None
+        self._list_gamma=None
+        self._list_dn=None
+
+    @property
+    def list_azimuth_time(self):
+        '''Returns the list of azimith time'''
+        if self._list_azimuth_time is None:
+            self._list_azimuth_time=self._parse_vectorlist('calibrationVectorList','azimuthTime','datetime')
+        return self._list_azimuth_time
```

Comment:
With `dataclass` use:
* the `__init__` would not be necessary because `dataclass` makes an init for you
* `@property` and load-on-access would not be necessary because attributes populated on init. I'm not a fan of this load-on-access pattern because it introduces a lot code repetition.
* the not really hidden `_<attribute>`s would not be necessary.

Reply:
The classes are re-implemented to `dataclass`